server/utils.go

```diff
@@ -24,3 +26,66 @@ func getGPRCVersion() (string, error) {
 	}
 	return out, nil
 }
+
+func setResourcesCPU(limits, requests float64, g generate.Generator) error {
+	if requests > limits {
+		return fmt.Errorf("CPU.Requests should not be greater than CPU.Limits")
+	}
+
+	// TODO(hmeng): this can be set as a global opt.
+	// by default, a pod sandbox is allowed to utilize one CPU core.
+	var cores float64 = 1
+
+	if limits != 0 || requests != 0 {
+		if limits > requests {
+			cores = limits
+		} else {
+			cores = requests
+		}
+	}
+	// by default, cpu.cfs_period_us is set to be 1000000 (i.e., 1s).
+	period := uint64(1000000)
+	quota := uint64(float64(period) * cores)
+
+	// the lower limit of cpu.cfs_quota_us is 1000.
+	if quota < 1000 {
+		quota = 1000
+	}
+
+	// the upper limit of cpu.cfs_quota_us is 1000000.
+	// adjust quota and period for the case where multiple CPUs are requested
+	// so that cpu.cfs_quota_us <= 1000000.
+	for quota > 1000000 {
```

Comment:
those numbers can be better placed at the beginning of this file with `const` - this may be `CfsQuotaUsMax` for instance